tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -44,10 +45,33 @@
 EXAMPLES_KEY = 'examples'
 
 
-def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int]) -> int:
+def _PartitionFn(record: bytes, num_partitions: int, buckets: List[int], feature_name: Text) -> int:
   assert num_partitions == len(
       buckets), 'Partitions do not match bucket number.'
-  bucket = int(hashlib.sha256(record).hexdigest(), 16) % buckets[-1]
+
+  if feature_name:
+    # Use given feature for partitioning the examples.
+    # Deserialize the record to tf.train.Example. 
+    example = tf.train.Example()
+    example.ParseFromString(record)
+    feature_list = example.features.feature[feature_name]
+    
```

Comment:
Note that this is a [Feature](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/example/feature.proto#L89) proto (which is a single feature, but this feature is a list), not FeatureList proto  (list of features)